src/tools/immutable_object_cache/Types.cc

```diff
@@ -101,6 +101,31 @@ void ObjectCacheReadData::decode_payload(bufferlist::const_iterator i) {
   ceph::decode(snap_id, i);
   ceph::decode(oid, i);
   ceph::decode(pool_namespace, i);
+  const int throttle_version[3] = {16, 0, 0};
+  int current_version[3];
+  if ((current_version[0] > throttle_version[0]) ||
+      (current_version[0] == throttle_version[0] &&
+      current_version[1] > throttle_version[1]) ||
+      (current_version[0] == throttle_version[0] &&
+      current_version[1] == throttle_version[1] &&
+      current_version[2] >= throttle_version[2]))
+    ceph::decode(object_size, i);
+}
+
+void ObjectCacheReadData::ceph_version_to_int(std::string version_string,
+                                              int version_int[])
+{
+  int tmp = 0;
+  for (int i = 0; i < 3; ++i) {
+    for (int j = 0; j < version_string.size(); ++j) {
+      if (version_string[i] >= '0' && version_string[i] <= '9') {
+        tmp = tmp * 10 + version_string[i];
+      } else {
+        version_int[i] = tmp;
+        tmp = 0;
+      }
+    }
+  }
```

Comment:
I don't know if the number after 16.0.0 will increase with the version. I just compared the first three numbers with 16.0.0. What do you think should be done here?The next line is a normal version number:
#define CEPH_GIT_NICE_VER "16.0.0-4331-g46abf2b154"

Reply:
ceph_version is string type, I change it to "int", extract three numbers from ceph version string and compare them with 16.0.0. Or is there a better way?